### PR TITLE
Make benchmark results more stable

### PR DESCRIPTION
**Pull request by kevingosse**
This PR solves 2 issues:

 - A default instance of the Tracer is automatically created when the static constructor runs. That instance runs things in the background, adding noise to benchmarks
 - Because the first invocation is much longer than the others, BenchmarkDotNet overestimates the time it takes to run the aspnetcore benchmark and ends up doing only one invocation. Running the benchmark once in the static constructor hides the warmup time from BenchmarkDotNet (I shouldn't have to do that, but I couldn't find a better solution)

### CODE REVIEW

File: src/Datadog.Trace/Configuration/TracerSettings.cs [hunk 1]
```diff
@@ -270,6 +270,8 @@ public TracerSettings(IConfigurationSource source)
         /// </summary>
         public bool StartupDiagnosticLogEnabled { get; set; }
 
+        internal static bool DisableSharedInstance { get; set; }
```
Comment (colin-higgins) [inline src/Datadog.Trace/Configuration/TracerSettings.cs:273]: Can we add a comment describing intent of use?

Comment (kevingosse) [reply src/Datadog.Trace/Configuration/TracerSettings.cs:277]: Sure!

File: benchmarks/Benchmarks.Trace/AspNetCoreBenchmark.cs [hunk 2]
```diff
@@ -25,8 +25,7 @@ public class AspNetCoreBenchmark
 
         static AspNetCoreBenchmark()
```
Comment (zacharycmontoya) [inline benchmarks/Benchmarks.Trace/AspNetCoreBenchmark.cs:26]: Would setting the warmup count property outlined here help counter the first request issue you're trying to solve here? https://benchmarkdotnet.org/articles/guides/choosing-run-strategy.html

Comment (kevingosse) [reply benchmarks/Benchmarks.Trace/AspNetCoreBenchmark.cs:26]: Unfortunately it changes the number of times the "warmup" step will be run, but not the number of invocations per step.

After further research, this is a known issue: https://github.com/dotnet/BenchmarkDotNet/issues/994
which seems to occur for applications emitting dynamic IL (like us): https://github.com/dotnet/BenchmarkDotNet/issues/837

The recommended workaround is to run an iteration of the benchmark in the setup, so I'll stick to that solution 🙂

**Review by zacharycmontoya — APPROVED** [review]

File: src/Datadog.Trace/Configuration/TracerSettings.cs (hunk 1 above)
Comment (lucaspimentel) [reply src/Datadog.Trace/Configuration/TracerSettings.cs:277]: Just saw this. I would have suggested `SharedInstanceEnabled` to keep the `*Enabled` pattern. Luckily it's internal so we can change it anytime 😉

**Pull request merged**